transaction-status: Remove conversions between spl re-exports and local sdk

joncinque opened the pull request:
#### Problem

While working on #33453, I noticed that helpers like `convert_pubkey` were still being used, even though #31607 removed the additional sdk crates that were pulled in through the SPL crates.

#### Summary of Changes

Remove all of the helpers! Essentially, I just ran a few commands to remove the functions and their usages:

```
git g -l convert_pubkey | xargs sed -i'' -re 's/convert_pubkey\(([^)]+)\)/\1/g'
```

And

```
git g -l convert_compiled_instruction | xargs sed -i'' -re 's/convert_compiled_instruction\(([^)]+)\)/\1/g'
```

And then cleaned up the build and clippy errors. Note that this only impacts tests, and nothing else.

Fixes #
<!-- OPTIONAL: Feature Gate Issue: # -->
<!-- Don't forget to add the "feature-gate" label -->

CriesofCarrots reviewed (dismissed):
😍 Thank you! Sorry I missed these when I did #31845. I guess I was a bit 🙈 to test code.

CriesofCarrots commented:
Wait, before you merge... I think you added a couple `convert_compiled_instruction()`s in #33453

joncinque commented:
> Wait, before you merge... I think you added a couple convert_compiled_instruction()s in https://github.com/solana-labs/solana/pull/33453

That's correct, thanks for noticing! Rebased and fixed those up in the last two commits
